trio/_highlevel_socket.py

```diff
@@ -119,6 +119,8 @@ def __init__(self, socket):
                         total_sent += sent
 
     async def wait_send_all_might_not_block(self):
+        if self.socket.did_shutdown_SHUT_WR:
+            raise trio.ClosedResourceError("can't send data after sending EOF")
```

Comment:
i'm not exactly sure where these changes fit in to the issues this MR is fixing. perhaps there is something that can be added to the newsfragment? it seems related related to what you were fixing in #1180 but it's in a different class.

Reply:
I don't remember exactly anymore, but I think the sequence was (1) I reworked the `send_eof` tests in `trio.testing.check_*`, (2) `SocketStream` started failing the more rigorous tests, (3) I fixed `SocketStream` to pass the tests again. I'll add a newsfragment...

Reply:
I don't remember exactly anymore, but I think the sequence was (1) I reworked the `send_eof` tests in `trio.testing.check_*`, (2) `SocketStream` started failing the more rigorous tests, (3) I fixed `SocketStream` to pass the tests again. I'll add a newsfragment...